src/cast_core/src/CrudActor.cpp

```diff
@@ -1206,21 +1296,125 @@ struct CrudActor::PhaseConfig {
                                    id);
         };
 
-        operations = phaseContext.getPlural<std::unique_ptr<BaseOperation>>(
-            "Operation", "Operations", addOperation);
+        auto addState =
+            [&](const Node& node,
+                const std::unordered_map<std::string, int> states) -> std::unique_ptr<State> {
+            //           State myState;
+            auto stateName = node["Name"].to<std::string>();
+            // Skipping repeat for now
+            // operations
+            auto stateOperations = node.getPlural<std::unique_ptr<BaseOperation>>(
+                "Operation", "Operations", addOperation);
+            // Transitions
+            std::vector<double> weights;
+            std::vector<Transition> transitions;
+            for (auto [k, transitionYaml] : node["Transitions"]) {
+                weights.emplace_back(transitionYaml["Weight"].to<double>());
+                transitions.emplace_back(Transition{
+                    states.at(transitionYaml["To"].to<std::string>()),
+                    Delay(transitionYaml["SleepBefore"], GeneratorArgs{phaseContext.rng(id), id})});
+            }
+            return std::unique_ptr<State>(new State{
+                std::move(stateOperations), stateName, std::move(weights), std::move(transitions)});
+        };
+        // Check if we have Operations or States. Through an error if we have both.
+        if (phaseContext["Operations"] && phaseContext["States"]) {
+            BOOST_THROW_EXCEPTION(InvalidConfigurationException(
+                "CrudActor has Operations and States at the same time."));
+        }
+        if (phaseContext["Operations"] || phaseContext["Operation"]) {
+            if (phaseContext["Continue"]) {
+                BOOST_THROW_EXCEPTION(
+                    InvalidConfigurationException("Continue option not valid if not using States"));
+            }
+            operations = phaseContext.getPlural<std::unique_ptr<BaseOperation>>(
+                "Operation", "Operations", addOperation);
+        } else if (phaseContext["States"]) {  // Parse out the states}
+            // build the list of states, and then actuall process the states
+            int i = 0;
+            std::unordered_map<std::string, int> stateNames;
+            for (auto [k, state] : phaseContext["States"]) {
+                stateNames.emplace(state["Name"].to<std::string>(), i);
```

Comment:
Maybe this is where the state name validation needs to go. must be present, non-blank, unique etc ...